controllers/nstemplateset/nstemplateset_controller.go

```diff
@@ -134,14 +134,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		logger.Error(err, "failed to either provision or update cluster resources")
 		return reconcile.Result{}, err
 	} else if createdOrUpdated {
-		return reconcile.Result{}, nil // wait for cluster resources to be created
+		// set the provisioned templates in the status as refs
+		return reconcile.Result{}, r.status.updateStatusClusterResourcesRevisions(ctx, nsTmplSet) // wait for cluster resources to be created
 	}
 
 	if createdOrUpdated, err := r.namespaces.ensure(ctx, nsTmplSet); err != nil {
 		logger.Error(err, "failed to either provision or update user namespaces")
 		return reconcile.Result{}, err
 	} else if createdOrUpdated {
-		return reconcile.Result{}, nil // something in the watched resources has changed - wait for another reconcile
+		return reconcile.Result{}, r.status.updateStatusNamespacesRevisions(ctx, nsTmplSet) // something in the watched resources has changed - wait for another reconcile
```

Comment:
IIRC, the current logic updates the label at the namespace only when it's done with the deletion, update, and creation of all inner resources.
We should follow the same approach. We should not update the revisions if we are still not completely done.
I understand that before refactoring the controller and merging the logic together, it can be bit weird/tricky, but I would move this out of this if statement, so it's updated when it's done

Reply:
yes, if I'm understanding this correctly it happens here: https://github.com/mfrancisc/member-operator/blob/c616b3f116561952151101c704da19173ebce3f0/controllers/nstemplateset/namespaces.go#L225-L229

and it's when actually it returns `createdOrUpdated` set to `true` here: https://github.com/mfrancisc/member-operator/blob/c616b3f116561952151101c704da19173ebce3f0/controllers/nstemplateset/namespaces.go#L73

We can move it outside anyways but wanted to update the revision only if there is a change.

Reply:
I've moved the status updated outside the if statements as discussed in the synch call.